### ISSUE TITLE
Logical Load Balancer JSON: the port should be optional

### ISSUE THREAD

**Issue by guillierf** [cloudfoundry/bosh-vsphere-cpi-release#255]
The Logical Load Balancer field is used to specify where the different PKS instances should land in term of NSX-T LB pool members.

When a NSX-T VS is created with multiple TCP ports, the servers created in the Pool Member field MUST have an empty field for the colum "Port"

This config is not possible today with Ops Mgr 2.7 or 2.8 release.
Ops Mgr mandate that in the JSON, the port field MUST be specified with a value.
Otherwise it will throw an error saying the field port is mandatory.

For our case, we simple want the Port field to be optional

**Comment by guillierf** [comment]
tested the CPI provided by Tushar and things work as expected:
-put a dummy Port on Ops Manager (to bypass the error message if the Port is not specified)
-the CPI created corrected the server pool members (3 in my case)

just 1 note: the server pool members were created with the IP field only. the name is empty. It may be good to fill the name field as well.

**Comment by julian-hj** [comment]
This looks like it might be a prerequisite to https://github.com/pivotal-cf/ops-manager/issues/411 and we already have most of an implementation. I will close and re-open it to get a story into the anycloud tracker backlog.

**Comment by julian-hj** [comment]
Reopening

**Comment by markstokan** [comment]
Closing.

The BOSH vSphere CPI code has always handled the absence of a port parameter gracefully.  We have added [tests](https://github.com/cloudfoundry/bosh-vsphere-cpi-release/commit/0ce6fc6abc7a3dc0e9237efc58550d4fc5a917ce) using the manager and policy api to draw attention to this.

An [issue](https://github.com/pivotal-cf/ops-manager/issues/411) is open with the Ops Manager team to make the Port optional when configuring this through the Resource Config UI.
